packages/alpinejs/src/evaluator.js

```diff
@@ -63,15 +63,23 @@ function generateFunctionFromString(expression) {
             ? `(() => { ${expression} })()`
             : expression
 
-    let func = new AsyncFunction(['__self', 'scope'], `with (scope) { __self.result = ${rightSideSafeExpression} }; __self.finished = true; return __self.result;`)
+    const safeAsyncFunction = () => {
+        try {
+            return new AsyncFunction(['__self', 'scope'], `with (scope) { __self.result = ${rightSideSafeExpression} }; __self.finished = true; return __self.result;`)
+        } catch ( error ) {
+            handleError( error, el, expression, false )
```

Comment:
What's the reason for this swallowing the error?

if the error in `x-data` is not thrown Devtools won't pick it up

Reply:
> > In those cases I think it was causing crazy down the line side effects as well that were essentially bogus errors because future lifecycle stuff like injecting magics depended on defined data. That's based on my foggy memory since it was last week but I'll know more once I test it a bit more again.
> 
> @danddanddand the way this was dealt with in v2 was to intercept all errors and re-throw them out of band using setTimeout, see:
> 
> https://github.com/alpinejs/alpine/blob/2.x/src/index.js#L94-L98

Okay we will try that

Reply:
Made a change that should address this now